src/Core/Utils/Attribs.hpp

```diff
@@ -290,8 +278,8 @@ class AttribManager {
     }
 
     /// Remove attribute by handle, invalidate the handles to this attribute.
-    /// \note: If a new attribute is added, old invalidated handles may lead to
-    ///        the new attribute.
+    /// \warning If a new attribute is added, old invalidated handles may lead to
```

Comment:
This is not clear. 
What is a old invalidated handle ?
How can two different handles lead to the same attribute ?

Reply:
*old invalidated handles* refers to handles to removed attributes. And since Handles only store an index, nothing prevents:
```c++
AttribManager m;
auto h = m.addAttrib<float>( "f_attr" );
m.removeAttrib( h ); // h is invalid in the sense that no attrib exists at its index, but h.isValid() == true
auto h2 = m.addAttrib<float>( "f_attr" ); // here h and h2 have the same index
{
    auto& attr2 = m.getAttrib( h2 ); // ok since its the right handle
    auto& attr = m.getAttrib( h ); // still ok while should not, nothing to forbid it
}
m.removeAttrib( h ); // still ok...
auto h3 = m.addAttrib<Vector2>( "v2_attr" ); // h, h2, and h3 have the same index
{
    auto& attr3 = m.getAttrib( h3 ); // ok since its the right handle
    auto& attr = m.getAttrib( h ); // compilation error for type mismatch
}
```

Reply:
Hum, this is not nice, or at least confusing...

So, to rephrase: the handlers are not persistent, and their validity is affected by any change in the manager. More, a handler cannot know if it is valid or not, because it doesn't know the state of the manager.

To fix that, I suggest we add a method 
```c++
template<typename T>
bool AttribManager::isValid(const AttribHandle<T>& in) const;
```
which check that `in` **looks** valid, in the sense that its name and type are the same than what is stored in the manager (this check needs to be carefully described in the documentation). This means that a handle cannot tell by itself if it his valid or not, but the manager can. If we do that, we also need to remove
```c++
template <typename T> class AttribHandle {
constexpr bool isValid() const { return m_idx.isValid(); }
}
```

Using such mechanism, you should be able to do:
```c++
AttribManager m;
auto h = m.addAttrib<float>( "f_attr" );
static_assert(m.isValid(h), "h is supposed to be valid");
m.removeAttrib( h ); // h is invalid
static_assert(! m.isValid(h), "h is supposed to be invalid");
auto h2 = m.addAttrib<float>( "f_attr" ); // here h and h2 have the same index
{
    auto& attr2 = m.getAttrib( h2 ); // ok since its the right handle
    auto& attr = m.getAttrib( h ); // still ok, because names and types are the same
    static_assert(m.isValid(h), "h is supposed to be valid");
}
m.removeAttrib( h ); // both h and h2 are invalid
static_assert(! m.isValid(h), "h is supposed to be invalid");
auto h3 = m.addAttrib<Vector2>( "v2_attr" );
static_assert(! m.isValid(h), "h is supposed to be invalid");
{
    auto& attr3 = m.getAttrib( h3 ); // ok since its the right handle
}
```

Reply:
To prevent it, we could do:
```c++
template <typename T>
void removeAttrib( AttribHandle<T>& h ) {
    const auto& att = getAttrib( h );
    removeAttrib( att.getName() );
    h.idx = Index::invalid();
}
```

Reply:
yes, handles validity should be defined by the manager, which can also prevent:
```c++
AttribManager m1, m2;
auto h1 = m1.addAttrib<float>( "f_attr1" );
auto h2 = m2.addAttrib<float>( "f_attr2" );
m1.removeAttrib( h1 );
if ( m1.isValid( h2 ) )
{
    m1.getAttrib( h2 );
...
}
```

Reply:
Ok for your second comment.

Reply:
make removeAttrib take a ref then? it is a bit safer, though it nothing prevents copying the handle beforehand and then use the copy

Reply:
I'm ok to remove the method `removeAttrib` by name, and keep only the one accepting a handle.

Reply:
What about the isValid method?

Reply:
We use the one I suggested, implemented by `AttribManager` as described above.